pkgs/tools/system/btop/default.nix

```
@@ -14,14 +17,36 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-uKR1ogQwEoyxyWBiLnW8BsOsYgTpeIpKrKspq0JwYjY=";
   };
 
+  nativeBuildInputs = lib.optionals stdenv.isDarwin [ gcc11 ];
+
+  hardeningDisable = lib.optionals (stdenv.isAarch64 && stdenv.isDarwin) [ "stackprotector" ];
+
+  ADDFLAGS = with darwin.apple_sdk;
+    lib.optional stdenv.isDarwin
+      "-F${frameworks.IOKit}/Library/Frameworks/";
+
+  buildInputs = with darwin.apple_sdk;
+    lib.optionals stdenv.isDarwin [
+      frameworks.CoreFoundation
+      frameworks.IOKit
+    ] ++ lib.optional (stdenv.isDarwin && !stdenv.isAarch64) (
```

This is more in-line with what you want to say:
```suggestion
    ] ++ lib.optional (stdenv.isDarwin && stdenv.isx86_64) (
```